spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkMicroBatchStream.java

```diff
@@ -205,6 +205,11 @@ private boolean shouldProcess(Snapshot snapshot) {
             "Cannot process delete snapshot : %s. Set read option %s to allow skipping snapshots of type delete",
             snapshot.snapshotId(), SparkReadOptions.STREAMING_SKIP_DELETE_SNAPSHOTS);
         return false;
+      case DataOperations.OVERWRITE:
+        Preconditions.checkState(skipDelete,
+            "Cannot process overwrite snapshot : %s. Set read option %s to allow skipping snapshots of type overwrite",
+            snapshot.snapshotId(), SparkReadOptions.STREAMING_SKIP_DELETE_SNAPSHOTS);
+        return false;
       default:
```

Comment:
Should add a test for this new conf option

Reply:
I have added the Unit Test.

Reply:
I have added the Unit Test.